### PR TITLE
CB-5052 [2562 FR] Highlight the row where any field is currently selected

### PR DESCRIPTION
**Pull request by sergeyteleshev**

### CODE REVIEW

File: webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/DataGridSelection/useGridSelectionContext.tsx [hunk 1]
```diff
@@ -238,6 +238,10 @@ export function useGridSelectionContext(tableData: ITableData, selectionAction:
     }
   }
 
+  function hasSelectedInRow(rowIdx: number) {
+    return props.selectionAction.getFocusedElement()?.row.index === rowIdx;
+  }
```
Comment (Wroud) [inline webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/DataGridSelection/useGridSelectionContext.tsx]: please rename selected to focused, also seems like it's better to expose computed like focused element and in cell / header renderers use this information to detect if current cell / header is focused.
Also please note, that `selectionAction.getFocusedElement()`'s column / row is not the same as `column.idx` / `rowIdx` index in data grid you need to use cell (https://github.com/dbeaver/cloudbeaver/pull/2952/files#diff-064428d02a72e78faffcf66fbd166bc59e9692a8fb55ccf92565cc00ef13a62cR43)

**Review by Wroud — CHANGES_REQUESTED** [review]

**Review by Wroud — APPROVED** [review]

**Review by devnaumov — APPROVED** [review]

File: webapp/packages/plugin-data-spreadsheet-new/src/theme/styles.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+import type { ThemeSelector } from '@cloudbeaver/core-theming';
+
+import type Dark from './dark.module.scss';
+import type Light from './light.module.scss';
+
+export const DATA_GRID_BASE_STYLES: ThemeSelector = async theme => {
+  let styles: typeof Light & typeof Dark;
+  const dark = (await import('./dark.module.scss')).default;
+  const light = (await import('./light.module.scss')).default;
```
Comment (sergeyteleshev) [inline webapp/packages/plugin-data-spreadsheet-new/src/theme/styles.ts]: had preloaded all styles at one time, cause the theme did not apply until `<TableGrid>` component was fully rerendered (reload page or tab)

File: webapp/packages/plugin-data-spreadsheet-new/src/theme/dark.module.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+@import '@cloudbeaver/core-theming/src/styles/theme-dark';
+
+:global .#{$theme-class} {
```
Comment (devnaumov) [inline webapp/packages/plugin-data-spreadsheet-new/src/theme/dark.module.scss]: We deleted these files on purpose, we dont want to use them anymore. We should define all styles in DataGridTable.module.css. Probably we want to add light-dark() postcss mixin or something like this

**Review by devnaumov — DISMISSED** [review]

File: webapp/packages/core-blocks/src/AppRefreshButton.module.css [hunk 4]
```diff
@@ -8,7 +8,7 @@
 .button {
   color: white;
   font-weight: 500;
-  background-color: #2a7cb4;
+  background-color: var(--theme-primary);
```
Comment (Wroud) [inline webapp/packages/core-blocks/src/AppRefreshButton.module.css]: theme may be unavailable in this component, please keep inlined styles

**Review by Wroud — CHANGES_REQUESTED** [review]

File: webapp/packages/plugin-data-spreadsheet-new/src/theme/dark.module.scss (hunk 3 above)
Comment (sergeyteleshev) [reply webapp/packages/plugin-data-spreadsheet-new/src/theme/dark.module.scss]: I rolled back changes, so we don't have a themed data grid. It is not required for now

File: webapp/packages/core-blocks/src/AppRefreshButton.module.css (hunk 4 above)
Comment (sergeyteleshev) [reply webapp/packages/core-blocks/src/AppRefreshButton.module.css]: done

**Review by Wroud — APPROVED** [review]

**Pull request merged**